lib/Retcon/Network/Server.hs

```diff
@@ -434,56 +434,69 @@ processDiff
     -> DiffID
     -> D.Patch
     -> m ()
-processDiff store cfg diff_id a_diff = do
+processDiff store cfg diffID resolveDiff = do
   res <- runExceptT act
   case res of
     Left e -> liftIO . errorM logName $ e
     Right () -> return ()
   where
     act = do
-      liftIO . infoM logName $ "Resolving errors in diff " <> show diff_id
+      liftIO . infoM logName $ "Resolving errors in diff " <> show diffID
       conflict <- getConflict
 
-      let en = EntityName . T.decodeUtf8 $ conflict ^. diffEntity
-          ik = InternalKey en $ conflict ^. diffKey
-          a_patch = Patch Unamed a_diff
+      let en           = EntityName . T.decodeUtf8 $ conflict ^. diffEntity
+          ik           = InternalKey en $ conflict ^. diffKey
+          resolvePatch = Patch Unamed resolveDiff
 
       (policy, srcs) <- getSources en
 
       -- 0. Load and update the initial document.
-      initial <- liftIO $ fromMaybe (emptyDocument en "<initial>") <$>
-        lookupInitialDocument store ik
-      let initial' = patch policy a_patch initial
+      initial <-  liftIO $ fromMaybe (emptyDocument en "<initial>")
+              <$> lookupInitialDocument store ik
+      let initial' = patch policy resolvePatch initial
 
       -- 1. Apply the patch to all sources.
-      mapM_ (\src -> liftIO $ do
-        doc <- either (const initial') id <$> getDocument store ik src
-        setDocument store ik (src, patch policy a_patch doc)
-        ) srcs
+      forM_ srcs $ \src -> liftIO $ do
+        doc <-  either (const initial') id
+            <$> getDocument store ik src
+        setDocument store ik (src, patch policy resolvePatch doc)
```

Comment:
If `doc == initial'` here then we're applying the patch to the document twice. This is OK, but might not be in future (if I make aeson-diff check the context for fuzzy applying)